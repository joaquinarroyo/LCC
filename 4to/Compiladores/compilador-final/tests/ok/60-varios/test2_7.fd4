### Probamos ejemplo para Closure Conversion.

let x : Nat = print "Def x = " 1

let y : Nat = print "Def y = " (2 + x)

let f (y:Nat) : Nat = 1 + x

let suma (x y : Nat) : Nat = x + y

let suma5 : Nat -> Nat = suma 5

let res : Nat = print "Resultado es " (suma5 2)
