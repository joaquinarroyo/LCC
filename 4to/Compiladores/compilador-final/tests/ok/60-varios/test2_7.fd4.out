Def x = 1
Def y = 3
Resultado es 7
